repos/exporter/lib/referenceData/tableRenameHandler.ts

```diff
@@ -2,17 +2,36 @@ import { Handler } from 'aws-lambda';
 import { ZipperLambdaBody } from 'fdbt-types/integrationTypes';
 import 'source-map-support/register';
 import { getConnection, checkReferenceDataImportHasCompleted, deleteAndRenameTables } from './tableRenamer';
+import { getSsmValue, putParameter } from '../ssm';
 
 export const handler: Handler<ZipperLambdaBody> = async () => {
     console.log('triggered reference data tables rename lambda... ');
 
     const connection = await getConnection();
 
-    await checkReferenceDataImportHasCompleted('txcOperatorLine', connection);
-    await checkReferenceDataImportHasCompleted('nocTable', connection);
-    await checkReferenceDataImportHasCompleted('naptanStop', connection);
+    let disableRenamer: string | undefined = 'true';
+    try {
+        disableRenamer = await getSsmValue('/scheduled/disable-table-renamer');
+    } catch (error) {
+        if (error instanceof Error) {
+            throw new Error(`Failed to get parameter from ssm: ${error.stack || ''}`);
+        }
 
-    await deleteAndRenameTables(connection);
+        throw error;
+    }
 
-    await connection.end();
+    if (disableRenamer === 'false') {
+        await checkReferenceDataImportHasCompleted('txcOperatorLine', connection);
+        await checkReferenceDataImportHasCompleted('nocTable', connection);
+        await checkReferenceDataImportHasCompleted('naptanStop', connection);
+
+        await deleteAndRenameTables(connection);
+
+        await connection.end();
+    } else {
+        throw new Error(
+            'The SSM Parameter used to check for errors in the scheduled job has returned TRUE indicating an issue',
+        );
+    }
```

Comment:
Can we enclose this in a try/catch block so that in case of an exception the ssm param is still set to false. Preferably in a finally block. This is cause the value is reset only as this step and if there is uncaught exception it will impact the next day run.